pkg/resource/template/resource.go

```diff
@@ -174,12 +185,14 @@ func (t *TemplateResource) createStageFile() error {
 // overwriting the target config file. Finally, sync will run a reload command
 // if set to have the application or service pick up the changes.
 // It returns an error if any.
-func (t *TemplateResource) sync() error {
+func (t *TemplateResource) sync() (err error) {
 	staged := t.StageFile.Name()
 	if t.keepStageFile {
 		log.Info("Keeping staged file: " + staged)
 	} else {
-		defer os.Remove(staged)
+		defer func() {
+			err = os.Remove(staged)
```

Comment:
Can this Remove be successful in an error case?  If so, then won't this set the return value to nil?

Reply:
yup. that is a problem; it could be over-riding a valid `err`